stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/viewholder/internal/ChannelViewHolder.kt

```diff
@@ -267,13 +267,18 @@ internal class ChannelViewHolder @JvmOverloads constructor(
         // pending - if the sync status says it's pending
 
         val currentUserSentLastMessage = lastMessage.user.id == ChatDomain.instance().user.value?.id
+        if (!currentUserSentLastMessage) {
+            messageStatusImageView.setImageDrawable(null)
+            return
+        }
+
         val lastMessageByCurrentUserWasRead = channel.isMessageRead(lastMessage)
         when {
-            !currentUserSentLastMessage || lastMessageByCurrentUserWasRead -> {
+            lastMessageByCurrentUserWasRead -> {
                 messageStatusImageView.setImageDrawable(style.indicatorReadIcon)
             }
 
-            currentUserSentLastMessage && !lastMessageByCurrentUserWasRead -> {
+            !lastMessageByCurrentUserWasRead -> {
                 messageStatusImageView.setImageDrawable(style.indicatorSentIcon)
             }
 
```

Comment:
I don't get how the "else" branch is possible here if check `lastMessageByCurrentUserWasRead` first and then `!lastMessageByCurrentUserWasRead`.

Reply:
That's very true - I can remove that part, it might possibly have been a remnant of previous impl.

Reply:
Fixed now I think, can you check! :]